### DISCUSSION TITLE
Custom CSS Options like for Jellyfin-Web

### DISCUSSION THREAD

**Discussion by hofq** [jellyfin/jellyfin-vue#2199] [Features]
Design preferences can be very custom, thus it would be nice to have an option to insert your own CSS.

I don't know how the Architecture would allow that, but for me, it would be very helpful.

**Comment by ferferga** [comment]
Hello

As for now, my stance is that **this won't be aded**. Doing this itself it's easy, the thing is that:

* We should provide all appearance/theming settings to the user, not force the user to do CSS.
    * I'd rather have the focus to make everything configurable through global CSS variables and maximize CSS reuse (the approach that Tailwind-like syntax permits) than forcing us to think "let's add an specific class to this element so it's customizable by the user" for every PR or change.
* There will be incompatibilities with web
* Custom CSS sometimes have been breaking Jellyfin Web, with users opening issues not related to us, hence difficulting troubleshooting.
* Although to every user their own, custom CSS introduces inconsistencies that are avoided if everything is managed centrally.
* Custom CSS it's only compatible with web-based clients, which is something that confuses people (why my theme is not applying to Android TV?)
* Using Tailwind-like approach for classes will make it extremely difficult to customize the client without customizing the classes of the elements and rebuilding the client (at that point, it's better tyo introduce the changes yourself and maintain your own fork)

TLDR: Let's provide all major user customization options like colors, views, etc... While leaving the most creative approaches for rebuilding the client.
